mpf/devices/segment_display/segment_display_text.py

```diff
@@ -76,28 +76,36 @@ def _create_characters(cls, text: str, display_size: int, collapse_dots: bool, c
         """Create characters from text and color them.
 
         - Colors are used from the left to the right (starting with the first character).
-        - If colors are shorter than text the last color is repeated for text.
-        - The first color is used to pad the text to the left if text is shorter than the display - thus text is right
-          aligned.
         - Dots and commas are embedded on the fly.
+        - Text will be right aligned on the display, thus if text is shorter than display spaces will be padded before the text
+        - Provided colors are assumed to be for the text itself, thus the list of colors is "right aligned" too
+        - If list of colors is less than the display size the list will be extended with white as default color on the left endswith
+          so that the provided colors are for the text and not being used for invisible spaces
         """
         char_list = []
-        left_pad_color = colors[0] if colors else None
-        default_right_color = colors[len(colors) - 1] if colors else None
         uncolored_chars = cls._embed_dots_and_commas(text, collapse_dots, collapse_commas, use_dots_for_commas)
-        colors = colors[-len(uncolored_chars):]
-        for char_code, char_has_dot, char_has_comma in uncolored_chars:
-            color = colors.pop(0) if colors else default_right_color
-            char_list.append(DisplayCharacter(char_code, char_has_dot, char_has_comma, color))
 
-        # ensure list is the same size as the segment display (cut off on left or right justify characters)
-        current_length = len(char_list)
+        # Adujust the color array if needed
+        color_length = len(colors)
+        if color_length > display_size:
+            for _ in range(color_length - display_size):
+                colors.pop(0) # remove very left color of array if too long
+        elif color_length < display_size:
+            for _ in range(display_size - color_length):
+                colors.append(RGBColor("white")) # add default color on the left of the array if too few colors
+
+        # ensure list is the same size as the segment display (cut off on left if too long or right justify characters if too short)
+        current_length = len(uncolored_chars)
         if current_length > display_size:
             for _ in range(current_length - display_size):
-                char_list.pop(0)
+                uncolored_chars.pop(0) # remove very left char of array if too long
         elif current_length < display_size:
             for _ in range(display_size - current_length):
-                char_list.insert(0, DisplayCharacter(SPACE_CODE, False, False, left_pad_color))
+                uncolored_chars.insert(0, (SPACE_CODE, False, False))
+
+        for _ in range(len(uncolored_chars)):
```

Comment:
By convention, using the underscore as a variable in a loop is meant to indicate that the variable is never used, i.e. the loop code doesn't care which iteration it's on. If you are going to use that variable, a letter is preferred (either the generic `i` or the first letter of what's being iterated, in this case `c`).

Reply:
On second thought, that's not necessary because `uncolored_chars` is iterable, so you can use `enumerate()` to get both the index and the value (which saves time on all the lookups in DisplayCharacter).

```
for i, char in enumerate(uncolored_chars):
  color = colors[i]
  char_list.append(DisplayCharacter(char[0], char[1], char[2], color))
```